src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -215,6 +231,109 @@ private DocValueFormat[] getSortValueFormats(final SortAndFormats sortAndFormats
         return sortAndFormats == null ? null : sortAndFormats.formats;
     }
 
+    private ReduceableSearchResult reduceCollectorResults(List<ReduceableSearchResult> results) {
```

Comment:
What exactly are we trying to do here? Can you add a comment. What exactly this r.reduce method will do?

Reply:
ack, adding comment